sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/EventProcessorOptions.cs

```diff
@@ -0,0 +1,217 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Text;
+using Azure.Core;
+using Azure.Messaging.EventHubs.Consumer;
+using Azure.Messaging.EventHubs.Core;
+
+namespace Azure.Messaging.EventHubs.Primitives
+{
+    /// <summary>
+    ///   The set of options that can be specified when creating a <see cref="EventProcessor{TPartition}" />
+    ///   to configure its behavior.
+    /// </summary>
+    ///
+    public class EventProcessorOptions
```

Comment:
As a public-facing type that we expect developers to create, our guidelines discourage the use of initialization via constructors for types that may change - especially options.   The reasoning is that adding a new field isn't a breaking change, but having to change the constructor to include it would be.... unless you force a default parameter there,  which kind of defeats the purpose.

That, by the way, is the same reasoning behind why the checkpoint and ownership became mutable with a default constructor when we flipped them from `internal` to `public`.